tests/integration/framework/process/statestore/statestore.go

```diff
@@ -54,8 +54,6 @@ func New(t *testing.T, fopts ...Option) *StateStore {
 	socketFile := util.RandomString(t, 8)
 
 	require.NotNil(t, opts.statestore)
-	_, ok := opts.statestore.(state.TransactionalStore)
-	require.True(t, ok, "statestore must implement state.TransactionalStore")
 
```

Comment:
Should we add require checks here which tests that `opts.statestore` implement the relevant interfaces if `opts.queryFunc` or `opts.transactionalStoreMultiMaxSizeFn` are not nil?

Reply:
`opts.queryFunc` and `opts.transactionalStoreMultiMaxSizeFn` would only have been set if `opts.statestore` was created through the `NewQuerier` and `NewTransactionalMultiMaxSize` factories. These factories return `WrappedQuerier` and `WrappedTransactionalMultiMaxSize` respectively, which implement default/dummy `Query` and `MultiMaxSize` methods, so a check can be possibly seen as redundant.